modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/display/context/AnalyticsReportsDisplayContext.java

```diff
@@ -52,16 +58,31 @@ public AnalyticsReportsDisplayContext(
 			return _data;
 		}
 
-		_data = Collections.singletonMap(
-			"context",
-			Collections.singletonMap(
-				"analyticsReportsDataURL",
-				String.valueOf(
-					_getResourceURL("/analytics_reports/get_data"))));
-
+		_data = HashMapBuilder.<String, Object>put(
+			"analyticsReportsDataURL",
+			String.valueOf(
+				_getResourceURL("/analytics_reports/get_data"))
+		).put(
+			"isPanelStateOpen", () -> {
+				HttpServletRequest _httpServletRequest = PortalUtil.getHttpServletRequest(_renderRequest);
```

Comment:
We should never use PortalUtil, since this is a static class, that depends on an OSGi injection that could hapends or not, you should use portal instead

Reply:
Good to know!
Could you point me to a valid example? 🥲